picamera2/picamera2.py

```diff
@@ -148,8 +211,15 @@ def __init__(self, camera_num=0, verbose_console=None, tuning=None):
                 os.environ["LIBCAMERA_RPI_TUNING_FILE"] = tuning_file.name
         else:
             os.environ.pop("LIBCAMERA_RPI_TUNING_FILE", None)  # Use default tuning
-        self.camera_manager = libcamera.CameraManager.singleton()
+        if not Picamera2._cm:
+            Picamera2._cm = CameraManager()
+        self.notifyme_r, notifymew = os.pipe()
+        self.notifyme_rr = os.fdopen(self.notifyme_r, 'rb')
+        self.notifyme_w = os.fdopen(notifymew, 'wb')
```

Comment:
I felt slightly that `self.notifyme_w` should be the file descriptor from `os.pipe()`. That would name things a bit more consistently with `self.notifyme_r` which is also the file descriptor. I didn't quite get the name `notifyme_rr` - why `rr` and is there something more obvious?

Also I'm not sure it's worth doing an `fdopen` on the write file descriptor, because `handle_request()` can just use `os.write(self.cameras[c].notifyme_w, b"\x00")` which would do the "flush" automatically. If that all makes sense....!

Reply:
Yes. I think that looks good!